http/http.go

```diff
@@ -19,34 +22,68 @@ type Server struct {
 
 // Listen starts the http server on the given port.
 func (server *Server) Listen(port int) {
-	r := mux.NewRouter()
-	s := r.PathPrefix(indexPath)
-	s.HandlerFunc(createIndex(server.Index)).Methods(http.MethodPost)
+	router := mux.NewRouter()
+	router.HandleFunc("/{index}/{id}", createIndex(server.Index)).Methods(http.MethodPost)
+	router.HandleFunc("/{index}/{id}", showIndex(server.Index)).Methods(http.MethodGet)
 
-	srv := &http.Server{
-		Handler: r,
-		Addr:    "127.0.0.1:" + strconv.Itoa(port),
-	}
-	srv.ListenAndServe()
+	log.Fatal(http.ListenAndServe(":"+strconv.Itoa(port), router))
 }
 
 func createIndex(indexService service.IndexService) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		path := strings.Replace(r.URL.Path, indexPath, "", 1)
-		splitPath := strings.Split(path, "/")
-
-		if len(splitPath) < 2 || splitPath[1] == "" {
-			http.Error(w, "No index was specified.", http.StatusBadRequest)
```

Comment:
If there's no index specified the server will automatically respond 404 because there's no route that matches the path.